addons/slack/template/index.js

```diff
@@ -0,0 +1,115 @@
+const slack = require('./slack.js');
+const Sdk = require('@fusebit/add-on-sdk');
+const { WebClient } = require('@slack/web-api');
+const Superagent = require('superagent');
+
+module.exports = async (ctx) => {
+    const sdk = {
+        bot: new WebClient(ctx.configuration.slack_bot_access_token),
+        user: ctx.configuration.slack_user_access_token ? new WebClient(ctx.configuration.slack_user_access_token) : null,
+    };
+
+    // Event registration/deregistration support
+    if (ctx.url.endsWith('events/registration')) {
+        Sdk.debug('Received Events API registration request');
+        if (ctx.configuration.fusebit_events_registration && ctx.configuration.fusebit_events_token) {
+            Sdk.debug('Events API configuration is present');
+
+            // TODO What prevents someone who finds out this URL to hit it and
+            // do an unauthorized registration/deregistration.
+
+            let response;
+            try {
+                let body = {
+                    app: ctx.configuration.slack_app_id,
+                    team_id: ctx.configuration.slack_team_id,
+                    handler: Sdk.getSelfUrl(ctx) + '/events/ingest',
+                };
+                if (ctx.method === 'POST') {
+                    Sdk.debug('Sending registration POST request for', body);
+                    response = await Superagent.post(ctx.configuration.fusebit_events_registration)
+                        .set('Authorization', `Bearer ${ctx.configuration.fusebit_events_token}`)
```

Comment:
Note to self: what's the lifecycle of this auth token? How does it get renewed, reconfigured, etc?

Reply:
It's a 2-time-use token. It gets used once to register a function to receive notifications (during addon install), and once more to deregister a function (during uninstall)

Reply:
There is now a detailed comment in `/template/configure.js` about how all this works